src/System.Memory/tests/Memory/Retain.cs

```diff
@@ -16,11 +16,7 @@ public static void MemoryRetainWithoutPinning()
             int[] array = { 1, 2, 3, 4, 5 };
             Memory<int> memory = array;
             MemoryHandle handle = memory.Retain();
-            unsafe
-            {
-                int* pointer = (int*)handle.PinnedPointer;
-                Assert.True(pointer == null);
-            }
+            Assert.False(handle.HasPointer);
```

Comment:
You probably still want to also assert that Pointer == null, right?

Reply:
Why? HasPointer does this check already.

Assert.False(handle.HasPointer) === Assert.False(_pointer != null) === Assert.True(_pointer == null)

Am I missing something?

Reply:
> Am I missing something?

You're assuming you know what the implementation of HasPointer is.